src/formatDate.js

```diff
@@ -9,6 +9,32 @@
  */
 function formatDate(date, fromFormat, toFormat) {
   // write code here
+  const dateParts = date.split(fromFormat[3]);
+  const dateObj = {};
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    dateObj[fromFormat[i]] = dateParts[i];
+  }
+
+  if (dateObj.YYYY && toFormat.includes('YY')) {
+    dateObj.YY = dateObj.YYYY.slice(-2);
+  } else if (dateObj.YY && toFormat.includes('YYYY')) {
```

Comment:
Consider checking if 'YY' is part of 'dateObj' before slicing 'YYYY' to avoid potential undefined behavior if 'YYYY' is not in the object. Similarly, check if 'YYYY' is part of 'dateObj' before converting 'YY' to 'YYYY'.